polysemy-template.hsfiles

```diff
@@ -0,0 +1,155 @@
+{-# START_FILE package.yaml #-}
+name:                {{name}}
+version:             0.1.0.0
+#synopsis:
+#description:
+homepage:            https://github.com/{{github-username}}{{^github-username}}githubuser{{/github-username}}/{{name}}#readme
+license:             BSD3
+author:              {{author-name}}{{^author-name}}Author name here{{/author-name}}
+maintainer:          {{author-email}}{{^author-email}}example@example.com{{/author-email}}
+copyright:           {{copyright}}{{^copyright}}{{year}}{{^year}}2019{{/year}} {{author-name}}{{^author-name}}Author name here{{/author-name}}{{/copyright}}
+category:            {{category}}{{^category}}Web{{/category}}
+extra-source-files:
+- README.md
+
+ghc-options: -O2 -flate-specialise -fspecialise-aggressively
+
+default-extensions:
+  - DataKinds
+  - FlexibleContexts
+  - GADTs
+  - LambdaCase
+  - PolyKinds
+  - RankNTypes
+  - ScopedTypeVariables
+  - TypeApplications
+  - TypeOperators
+  - TypeFamilies
+  
+dependencies:
+  - base >= 4.7 && < 5
+  - polysemy
+  - polysemy-plugin
+
+executables:
+  main:
+    source-dirs:      src
+    main:             Main.hs
+
+{-# START_FILE Setup.hs #-}
+import Distribution.Simple
+main = defaultMain
+
+{-# START_FILE src/Main.hs #-}
+  {-# LANGUAGE TemplateHaskell #-}
+  module Main where
+  import Prelude hiding (throw, catch, bracket)
+  import Polysemy
+  import Polysemy.Input
+  import Polysemy.Output
+  import Polysemy.Error
+  import Polysemy.Resource
+  
+  import Teletype hiding (main)
```

Comment:
No point in defining main if we're just going to hide its import. Remove it from `Teletype`?

Reply:
I was mainly doing this in order to have something quickly buildable to tinker with. I left the definition in case someone wanted to swap the Teletype example in. I made the resource example the primary one just because it shows off a few more things than Teletype. 

Do you still think it should be removed? Or perhaps there's even better ways to promote "play" with the code?

Reply:
I agree now, better to use the imported version of Teletype. It's less maintenance work than having it in both places.